modules/ha-cluster/main.tf

```diff
@@ -0,0 +1,894 @@
+  -hcl='ports { server = 8300, serf_lan = 8301, serf_wan = 8302, https = 8501, grpc = 8502, grpc_tls = 8503 }' \
+%{endif~}
+%{if var.acls~}
+  -hcl='acl {enabled = true, default_policy = "deny", down_policy = "extend-cache", enable_token_persistence = true}' \
+  -hcl='acl = { tokens = { initial_management = "${local.bootstrap_token}", default = "${local.bootstrap_token}" }}' \
+%{endif~}
+%{if var.acls && local.enable_mesh_gateway_wan_federation~}
+  -hcl='acl = { enable_token_replication = true }' \
+%{endif~}
+%{if var.acls && var.replication_token != ""~}
+  -hcl='acl = { tokens = { replication = "${var.replication_token}"}}' \
+%{endif~}
+%{if var.primary_datacenter != ""~}
+  -hcl='primary_datacenter = "${var.primary_datacenter}"' \
+%{endif~}
+%{if var.aws_auto_join~}
+  -retry-join "provider=aws tag_key=Consul-Auto-Join tag_value=${var.name} service=ecs" \
+%{endif~}
+%{if length(var.retry_join_wan) > 0~}
+  -hcl='retry_join_wan = [
+  %{for addr in var.retry_join_wan~}
+    "${addr}",
+  %{endfor~}
+  ]' \
+%{endif~}
+%{if local.enable_mesh_gateway_wan_federation~}
+  -hcl='connect { enable_mesh_gateway_wan_federation = true }' \
+%{endif~}
+%{if length(var.primary_gateways) > 0~}
+  -hcl='primary_gateways = [
+  %{for addr in var.primary_gateways~}
+    "${addr}",
+  %{endfor~}
+  ]' \
+%{endif~}
+EOF
+  // Generate consul_count server configurations.
+  // The index of the range iteration is specified multiple times to be used the same number of times in the template.
+  //consul_server_commands = [for i in range(var.consul_count) : format(local.consul_server_command_template, i)]
+
+  // We use this command to generate the server certs dynamically before the servers start
+  // because we need to add the IP of the task as a SAN to the certificate, and we don't know that
+  // IP ahead of time.
+  consul_server_tls_init_command_template = <<EOF
+ECS_IPV4=$(curl -s $ECS_CONTAINER_METADATA_URI_V4 | jq -r '.Networks[0].IPv4Addresses[0]')
+cd /consul
+echo "$CONSUL_CACERT_PEM" > ./consul-agent-ca.pem
+echo "$CONSUL_CAKEY" > ./consul-agent-ca-key.pem
+consul tls cert create -server \
+  -node="${local.node_name}%s" \
+  -dc="${var.datacenter}" \
+  -additional-ipaddress=$ECS_IPV4 \
+  -additional-dnsname="${var.name}%s.${local.service_discovery_namespace}" \
+%{if length(var.additional_dns_names) > 0~}
+  %{for dnsname in var.additional_dns_names~}
+    -additional-dnsname="${dnsname}" \
+  %{endfor~}
+%{endif~}
+EOF
+
+  consul_portmap = [{
+    containerPort : 8300,
+    hostPort : 8300,
+    protocol : "tcp"
+    }, {
+    containerPort : 8301,
+    hostPort : 8301,
+    protocol : "tcp"
+    }, {
+    containerPort : 8302,
+    hostPort : 8302,
+    protocol : "tcp"
+    }, {
+    containerPort : 8500,
+    hostPort : 8500,
+    protocol : "tcp"
+    }, {
+    containerPort : 8501,
+    hostPort : 8501,
+    protocol : "tcp"
+    }, {
+    containerPort : 8502,
+    hostPort : 8502,
+    protocol : "tcp"
+    }, {
```

Comment:
I notice you have `ports.grpc_tls = 8503` above, so probably add 8503 to the list of container ports here for consistency. (Note that `ports.grpc_tls` is only supported in Consul 1.14+)